src/components/molecules/UiLoader/_internal/UiLoaderTransition.vue

```diff
@@ -21,49 +21,49 @@ import {
   onMounted,
 } from 'vue';
 import type { CSSProperties } from 'vue';
+import type { DefineAttrsProps } from '../../../../types';
 
-export type TransitionState = 'from' | 'active' | 'to';
-export type Move = 'leave' | 'enter';
-export type TransitionHook = `before-${Move}`
-  | `${Move}`
-  | `after-${Move}`
-  | `${Move}-cancelled`;
-const props = defineProps({
+export type LoaderTransitionState = 'from' | 'active' | 'to';
+export type LoaderTransitionMove = 'leave' | 'enter';
+export type LoaderTransitionHook = `before-${LoaderTransitionMove}`
+  | `${LoaderTransitionMove}`
+  | `after-${LoaderTransitionMove}`
+  | `${LoaderTransitionMove}-cancelled`;
+export interface LoaderTransitionProps {
   /**
    * Use this props to show UiLoader component.
    */
-  isLoading: {
-    type: Boolean,
-    default: true,
-  },
+  isLoading?: boolean;
   /**
    * Use this props to set transition name.
    */
-  name: {
-    type: String,
-    default: '',
-  },
+  name?: '';
   /**
    * Use this props to apply a transition on the initial render.
    */
-  appear: {
-    type: Boolean,
-    default: true,
-  },
+  appear?: boolean;
   /**
    * Use this props to use opacity transition type.
    */
-  isOpacityTransitionType: {
-    type: Boolean,
-    default: false,
-  },
+  isOpacityTransitionType?: boolean;
+}
+export type LoaderTransitionAttrsProps = DefineAttrsProps<LoaderTransitionProps>;
+export interface LoaderTransitionEmits {
+  (e: LoaderTransitionHook, el: HTMLElement): void;
+}
+
+const props = withDefaults(defineProps<LoaderTransitionProps>(), {
+  isLoading: true,
+  name: '',
+  appear: true,
+  isOpacityTransitionType: false,
 });
+const emit = defineEmits<LoaderTransitionEmits>();
 const loaderEl = ref<HTMLElement | null>(null);
 const contentEl = ref<HTMLElement | null>(null);
 const isLoaderVisible = ref(props.isLoading);
 const transitionEl = computed(() => (isLoaderVisible.value ? loaderEl.value : contentEl.value) as HTMLElement);
 const move = computed(() => (isLoaderVisible.value === props.isLoading ? 'enter' : 'leave'));
```

Comment:
What do you think about creating const variables for these `'enter'` and `'leave'` and reusing them both here and in the type declaration? 
```typescript
const ENTER_MOVE = 'enter';
const LEAVE_MOVE = 'leave';
export type LoaderTransitionMove = typeof ENTER_MOVE | typeof LEAVE_MOVE;
...
const move = computed<LoaderTransitionMove>(() => (isLoaderVisible.value === props.isLoading ? ENTER_MOVE : LEAVE_MOVE));
```

Reply:
great idea 👍